forest/bonsai/simulate_gps_data.py

```diff
@@ -1317,6 +1303,15 @@ def sim_gps_data(
         ValueError: if Overpass query fails
     """
 
+    # check if api key is valid
```

Comment:
Isn't this already checked in `get_path()`?

Reply:
Yes but I thought if we want it to exit right away in case of API key not working we should put in the beginning. Otherwise until the `get_path` function is called there are 2 api calls to OSM that may take some time themselves. If you think it is not necessary to prompt an error right away in case of wrong API key I can remove it.

Reply:
Where are those calls and why would they take a long time to fail? If the key is invalid then any request should fail very quickly.

Reply:
It's not that they take a long time to fail, they won't fail, it just takes some time to pull the results. They are the ones with the OSM API not the ORS one. There is one on line 1341 to generate random addresses and one on line 1377 to generate nearby locations with attributes.